array.go

```diff
@@ -360,6 +360,97 @@ func (a *Array) NonEmptyDomain() ([]NonEmptyDomain, bool, error) {
 	}
 }
 
+// NonEmptyDomainVarFromName retrieves the non-empty domain from an array for a
+// given dimension name.
+// This returns the bounding coordinates for the dimension
+func (a *Array) NonEmptyDomainVarFromName(dimName string) ([]NonEmptyDomain, []NonEmptyDomain, bool, error) {
+
+	schema, err := a.Schema()
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	domain, err := schema.Domain()
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	domainType, err := domain.Type()
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	cDimName := C.CString(dimName)
+	defer C.free(unsafe.Pointer(cDimName))
+
+	var cstartSize C.uint64_t
+	var cendSize C.uint64_t
+
+	var isEmpty C.int32_t
+
+	var start interface{}
+	var end interface{}
+	var cstart unsafe.Pointer
+	var cend unsafe.Pointer
+
+	switch domainType {
+	case TILEDB_STRING_ASCII:
+		ret := C.tiledb_array_get_non_empty_domain_var_size_from_name(
```

Comment:
I think everything can be moved outside the switch case here? There is nothing specific to strings?

Reply:
moved in merged branch